cdc: TestChangefeedDescription test panic

Seen in CI (https://teamcity.cockroachdb.com/buildConfiguration/Cock:roach_UnitTests_BazelUnitTests/8078031?showRootCauses=false&expandBuildChangesSection=true&expandBuildTestsSection=true)

```
=== RUN   TestChangefeedDescription/CREATE_CHANGEFEED_INTO_$1_WITH_updated,_envelope_=_$2,_schema_change_policy='stop'_AS_SELECT_a_FROM_foo_WHERE_status_IN_('open',_'closed')
panic: expected *DBool, found tree.dNull
goroutine 38622 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDBool({0x6c88718?, 0x9e02860?})
  github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:268 +0x93
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Eval(0xc005d06b40, {0x6c69ff0, 0xc001c33a00}, {0xc003c57260, {0x17334d4143526b87, 0x0, 0x0}, {0xc00a4ac840, 0x2, 0x2}, ...}, ...)
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:119 +0x3f2
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).ConsumeEvent(0xc0068416c0, {0x6c69ff0, 0xc001c33a00}, {0xc00ce05110, 0x2, {0x17334d417dce4931, 0x0, 0x0}, {0xc0e164b7a00b45ce, 0x856add03e, ...}, ...})
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:337 +0x98c
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0xc002651500, {0x6c69ff0, 0xc001c33a00}, {0x6c48d10, 0xc0068416c0}, 0x3)
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:604 +0x259
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x6c69ff0?, 0xc001c33a00?})
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:576 +0x3c
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
  golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
  golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa5

```

Jira issue: CRDB-22734

--- a/pkg/ccl/changefeedccl/cdceval/expr_eval.go
+++ b/pkg/ccl/changefeedccl/cdceval/expr_eval.go
@@ -116,7 +116,11 @@ func (e *Evaluator) Eval(
 	case err := <-e.errCh:
 		return cdcevent.Row{}, err
 	case row := <-e.rowCh:
-		if !tree.MustBeDBool(row[0]) {
+		filter, err := tree.GetBool(row[0])
+		if err != nil {
+			return cdcevent.Row{}, err
+		}
+		if !filter {
 			// Filter did not match.
 			return cdcevent.Row{}, nil
 		}

--- a/pkg/ccl/changefeedccl/cdceval/expr_eval_test.go
+++ b/pkg/ccl/changefeedccl/cdceval/expr_eval_test.go
@@ -56,10 +56,11 @@ CREATE TABLE foo (
   e status DEFAULT 'inactive',
   f STRING,
   g STRING,
+  flag BOOL,
   PRIMARY KEY (b, a),
   FAMILY main (a, b, e),
   FAMILY only_c (c),
-  FAMILY f_g_fam(f,g)
+  FAMILY f_g_fam(f, g, flag)
 )`)
 	desc := cdctest.GetHydratedTableDescriptor(t, s.ExecutorConfig(), "foo")
 
@@ -263,7 +264,7 @@ CREATE TABLE foo (
 			},
 		},
 		{
-			testName:   "main/concat",
+			testName:   "concat",
 			familyName: "f_g_fam",
 			actions: []string{
 				"INSERT INTO foo (a, b, f) VALUES (42, 'concat', 'hello')",
@@ -278,6 +279,32 @@ CREATE TABLE foo (
 				{
 					keyValues: []string{"concat", "42"},
 					allValues: map[string]string{"a": "42", "b": "concat", "ff": "hellohello", "gg": "NULL"},
+				},
+			},
+		},
+		{
+			testName:   "boolean flag",
+			familyName: "f_g_fam",
+			actions: []string{
+				"INSERT INTO foo (a, b, g, flag) VALUES (42, 'true', 'gee', true)",
+				"INSERT INTO foo (a, b, g, flag) VALUES (42, 'false', 'gee', false)",
+				"INSERT INTO foo (a, b, g) VALUES (42, 'maybe', 'gee')", // NULL flag should be treated as false.
+			},
+			stmt:             "SELECT a, b, flag FROM foo WHERE flag",
+			expectMainFamily: repeatExpectation(decodeExpectation{expectUnwatchedErr: true}, 3),
+			expectFGFamily: []decodeExpectation{
+				{
+					keyValues:      []string{"false", "42"},
+					expectFiltered: true,
+				},
+				{
+					// null boolean value should also be filtered.
+					keyValues:      []string{"maybe", "42"},
+					expectFiltered: true,
+				},
+				{
+					keyValues: []string{"true", "42"},
+					allValues: map[string]string{"a": "42", "b": "true", "flag": "true"},
 				},
 			},
 		},

